preference-v7/src/main/java/com/takisoft/fix/support/v7/preference/EditTextPreference.java

```diff
@@ -39,4 +39,10 @@ public void setText(String text) {
             notifyChanged();
         }
     }
+
+    @Override
+    public CharSequence getSummary() {
+        CharSequence summary = super.getSummary();
+        return summary == null ? null : String.format(summary.toString(), getText());   // TODO: formatting?
+    }
```

Comment:
Yet another version I've been using for years, which allows both hint and formatted summary:

```
    @Override
    public CharSequence getSummary() {
        String text = getText();
        if (TextUtils.isEmpty(text)) {
            CharSequence hint = getEditText().getHint();
            if (!TextUtils.isEmpty(hint)) {
                return hint;
            } else {
                return super.getSummary();
            }
        } else {
            CharSequence summary = super.getSummary();
            if (!TextUtils.isEmpty(summary)) {
                return String.format(summary.toString(), text);
            } else {
                return summary;
            }
        }
    }
```